contributors/devel/sig-testing/monitoring.md

```diff
@@ -0,0 +1,264 @@
+results in a grid. TestGrid's back end components are open sourced and can be
+viewed here https://github.com/GoogleCloudPlatform/testgrid. The front-end code
+that renders the dashboard is not currently open sourced.
+
+The Kubernetes community has its own instance of TestGrid, https://testgrid.k8s.io/
+which we use to monitor and observe the health of the project.
+
+Each [SIG](https://github.com/kubernetes/community/blob/master/sig-list.md) has
+its own set of dashboards. Each dashboard is composed of different jobs (build,
+unit test, integration test, end-to-end (e2e) test, etc.) These views allow
+different teams to monitor and understand how their areas are doing.
+
+E2E jobs are in turn made up of test stages (e.g., bootstrapping a Kubernetes
+cluster, tearing down a Kubernetes cluster) and e2e tests are organized
+hierarchically per Component and Subcategory within that component. (e.g., the
+[Kubectl
+client](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go#L229)
+component has tests that describe the expected behavior of [Kubectl
+logs](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go#L1389).
+One such test is [should be able to retrieve and filter
+logs](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go#L1412)).
+
+We highly encourage SIGs to periodically monitor the dashboards related to the
+sub-projects that they own. If you see that a job or test has been failing,
+please raise an issue with the corresponding SIG in either their mailing list or
+in Slack.
+
+Help maintaining tests is always needed throughout the project:
+the following contributions are always welcome
+ - [Triage Flaky Tests](https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+is%3Aopen+kind%2Fflake)
+ - [Fix Flaky Test](https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+is%3Aopen+kind%2Fflake+-label%3Aneeds-triage+)
+ 
+**Note**: It is important that all SIGs periodically monitor their jobs and
+tests.
+
+Furthermore, if jobs or tests are failing or flaking, then pull requests will
+take a lot longer to be merged.
+
+### What dashboards should I monitor?
+
+This depends on what areas of Kubernetes you want to contribute to. You should
+monitor the dashboards owned by the SIG you are working with.
+
+Additionally, you should check:
+
+* https://testgrid.k8s.io/sig-release-master-blocking 
+* https://testgrid.k8s.io/sig-release-master-informing
+
+since these jobs run tests that are used by SIG Release to determine the overall
+quality of Kubernetes and whether or not the commit on master can be considered
+suitable for release. Failing tests on a job in sig-release-master-blocking
+block a release from taking place.
+
+Also, make sure to periodically check on the "blocking" and "informing"
+dashboards for past releases. See https://testgrid.k8s.io/sig-release
+
+---
+
+## What do I do when I see a TestGrid alert?
```

Comment:
I'm not sure if it makes sense in this doc, but here's something I do, and see others do, which might be just tribal vs. documented knowledge.

If a test fails on my PR, and it's clearly not related to the code I wrote, I try to find any open issues that appear related (have the name of the test in them, describe a similar error, etc.).  I'll link that in the comment I use to retrigger jobs, either calling the job out specifically:
```
./test pull-kubernetes-foo
https://github.com/kubernetes/kubernetes/issues/foo
```
or even if just spamming retest
```
./retest
https://github.com/kubernetes/kubernetes/issues/foo
```

I might even back-link from the issue to the PR that encountered it, to bump the issue's last updated date.

That way, the issue will accumulate links to other occurrences of the flake.  Lots of links = good signal this is actual pain contributors are regularly experiencing.

Reply:
Good stuff, I'll add this.